components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/RefreshGrantHandler.java

```diff
@@ -282,6 +297,63 @@ private boolean validateRefreshTokenStatus(RefreshTokenValidationDataDO validati
         return true;
     }
 
+    private boolean validateAuthenticatedUser(RefreshTokenValidationDataDO validationBean,
```

Comment:
What is the main purpose of this validation method?

Reply:
Validating the account lock status before issuing a new access token.

Reply:
Shall we improve the method name to reflect it?

Reply:
My thought was that there could be future validation related to the the authenticated user (Ex: Disable user). In that case, we could reuse this method. Does that make sense?

Reply:
ack. in that case shall we move account lock validation into a separate private method to improve the maintainability ?

Reply:
Done https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2581/commits/6bd8aaa877daabf354833109557bb9fb4dff26c9

Reply:
Let's only move the account lock validation to the new method.

Reply:
Done https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2581/commits/d82304261ad30a2ab0dd7190df7c09076ee8179b